common/infallible/src/math.rs

```diff
@@ -41,43 +42,54 @@
 #[macro_export]
 macro_rules! checked {
     ($a:tt + $b:tt) => {{
-        #![allow(clippy::integer_arithmetic)]
-        $a.checked_add($b).ok_or_else(|| format!("Operation results in overflow/underflow: {} + {}", $a, $b))
+        $a.checked_add($b).ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} + {}", $a, $b)))
     }};
     ($a:tt - $b:tt) => {{
-        #![allow(clippy::integer_arithmetic)]
-        $a.checked_sub($b).ok_or_else(|| format!("Operation results in overflow/underflow: {} - {}", $a, $b))
+        $a.checked_sub($b).ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} - {}", $a, $b)))
     }};
     ($a:tt * $b:tt) => {{
-        #![allow(clippy::integer_arithmetic)]
-        $a.checked_mul($b).ok_or_else(|| format!("Operation results in overflow/underflow: {} * {}", $a, $b))
+        $a.checked_mul($b).ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} * {}", $a, $b)))
     }};
     ($a:tt / $b:tt) => {{
-        #![allow(clippy::integer_arithmetic)]
-        $a.checked_div($b).ok_or_else(|| format!("Operation results in overflow/underflow: {} / {}", $a, $b))
+        $a.checked_div($b).ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} / {}", $a, $b)))
     }};
     ($a:tt + $($tokens:tt)*) => {{
         checked!( $($tokens)* ).and_then(|b| {
             b.checked_add($a)
-                .ok_or_else(|| format!("Operation results in overflow/underflow: {} + {}", b, $a))
+                .ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} + {}", b, $a)))
         })
     }};
     ($a:tt - $($tokens:tt)*) => {{
         checked!( $($tokens)* ).and_then(|b| {
             b.checked_sub($a)
-                .ok_or_else(|| format!("Operation results in overflow/underflow: {} - {}", b, $a))
+                .ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} - {}", b, $a)))
         })
     }};
     ($a:tt * $($tokens:tt)*) => {{
         checked!( $($tokens)* ).and_then(|b| {
             b.checked_mul($a)
-                .ok_or_else(|| format!("Operation results in overflow/underflow: {} * {}", b, $a))
+                .ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} * {}", b, $a)))
         })
     }};
     ($a:tt / $($tokens:tt)*) => {{
         checked!( $($tokens)* ).and_then(|b| {
             b.checked_div($a)
-                .ok_or_else(|| format!("Operation results in overflow/underflow: {} / {}", b, $a))
+                .ok_or_else(|| $crate::math::ArithmeticError(format!("Operation results in overflow/underflow: {} / {}", b, $a)))
         })
     }};
 }
+
+#[derive(Debug)]
+pub struct ArithmeticError(pub String);
```

Comment:
The field doesn't need to be public.

Reply:
Looks like it does, otherwise it complains for some reason that it cannot use the `String` constructor when constructing the error type like : `ArithmeticError(<string>)`, I guess this is because the actual invocation is done on the foreign crate when the macro is expanded.

Reply:
Oh that's right, i had forgotten that these were created in the macro, you can ignore this then :)